src/libraries/Common/tests/SingleFileTestRunner/SingleFileTestRunner.cs

```diff
@@ -74,10 +77,58 @@ public static int Main(string[] args)
                     noTraits.Add(traitKeyValue[0], values = new List<string>());
                 }
                 values.Add(traitKeyValue[1]);
+                i++;
             }
             if (args[i].Equals("-xml", StringComparison.OrdinalIgnoreCase))
             {
                 xmlResultFileName=args[i + 1].Trim();
+                i++;
+            }
+            if (args[i].Equals("-class", StringComparison.OrdinalIgnoreCase))
+            {
+                filters.IncludedClasses.Add(args[i + 1].Trim());
+                i++;
+            }
+            if (args[i].Equals("-noclass", StringComparison.OrdinalIgnoreCase) || args[i].Equals("-class-", StringComparison.OrdinalIgnoreCase))
+            {
+                filters.ExcludedClasses.Add(args[i + 1].Trim());
+                i++;
+            }
+            if (args[i].Equals("-method", StringComparison.OrdinalIgnoreCase))
+            {
+                filters.IncludedMethods.Add(args[i + 1].Trim());
+                i++;
+            }
+            if (args[i].Equals("-nomethod", StringComparison.OrdinalIgnoreCase) || args[i].Equals("-method-", StringComparison.OrdinalIgnoreCase))
+            {
+                filters.ExcludedMethods.Add(args[i + 1].Trim());
+                i++;
+            }
+            if (args[i].Equals("-namespace", StringComparison.OrdinalIgnoreCase))
+            {
+                filters.IncludedNamespaces.Add(args[i + 1].Trim());
+                i++;
+            }
+            if (args[i].Equals("-nonamespace", StringComparison.OrdinalIgnoreCase) || args[i].Equals("-namespace-", StringComparison.OrdinalIgnoreCase))
+            {
+                filters.ExcludedNamespaces.Add(args[i + 1].Trim());
+                i++;
+            }
+            if (args[i].Equals("-parallel", StringComparison.OrdinalIgnoreCase))
```

Comment:
Thank you for extending this! Especially the `-parallel` option is something that I've been doing locally pretty much forever.

I would like us to get rid of this runner eventually (#70432). The xunit runner from Arcade used to support single file testing (including remote executor) but all of that was deleted when UWP support was deleted from Arcade. I'd like to restore it at some point.

I wonder though whether crossgen2 testing needs the single file runner - could it get by with the real runner?

Reply:
I suspect it could use the normal file runner, but this looks like it will work, and I don't understand the libraries testing enough to understand how exactly to make this work with the normal runner without doing more research. With the additional arguments, I don't see a huge benefit to making this different from SingleFile or NativeAOT testing. As a side benefit, enabling R2R singlefile testing here becomes extremely easy should we want to do so.